src/sqlfluff/core/linter/linting_result.py

```diff
@@ -181,3 +184,32 @@ def tree(self) -> Optional[BaseSegment]:  # pragma: no cover
                 "path."
             )
         return self.paths[0].tree
+
+    def mark_failed_files_unfixable(self) -> Tuple[int, int]:
+        """Discard lint fixes for files with templating or parse errors.
```

Comment:
The function name and the description differs.

What about:

```suggestion
    def discard_unfixable_errors(self) -> Tuple[int, int]:
        """Discard lint fixes for files with templating or parse errors.
```

Reply:
It's a difficult function to name. The name you proposed is also inaccurate: we're not discarding _errors_, we're discarding _fixes_ for errors and keeping the actual errors.

How about `discard_fixes_for_lint_errors_in_files_with_templating_or_parsing_errors`?

Reply:
Renamed as `discard_fixes_for_lint_errors_in_files_with_tmp_or_prs_errors()`.

Reply:
A function name almost as long as your branch names 😁